File: JavaScript/Sleep Debt Calculator.js
const getSleepHours = (day) => {
day = day.toLowerCase();
  switch (day) {
    case "monday":
    return 8;
    break;

    case "tuesday":
    return 7;
    break;

    case "wednesday":
    return 6;
    break;

    case "thursday":
    return 5;
    break;

    case "friday":
    return 4;
    break;

    case "saturday":
    return 3;
    break;

    case "sunday":
    return 2;
    break;

    default:
    return "Invalid input";
  }
}

const getActualSleepHours = () =>
+  getSleepHours("Monday")
+  getSleepHours("Tuesday");
+  getSleepHours("Wednesday");
+  getSleepHours("Thursday");
+  getSleepHours("Friday");
+  getSleepHours("Saturday");
+  getSleepHours("Sunday");

//console.log(getActualSleepHours());

const getIdealSleepHours = (idealHours) => {
  return idealHours * 7;
}

//console.log(getIdealSleepHours());

const calculateSleepDebt = (ideal) => {
  var actualSleepHours = getActualSleepHours();
  var idealSleepHours = getIdealSleepHours(ideal);
  if (actualSleepHours === idealSleepHours) {
    console.log("You get the perfect amount of sleep");
  } else if (actualSleepHours > idealSleepHours) {
    var totalDifference = actualSleepHours - idealSleepHours;
    console.log("You get too much sleep");
    console.log(`You are ${totalDifference} hours ahead of your schedule`);
  } else {
    var totalDifference = idealSleepHours - actualSleepHours;
    console.log("You don't get enough sleep");
    console.log(`You are ${totalDifference} hours behind your schedule`);
  }

}

calculateSleepDebt(5);

File: Python/Sleep Debt Calculator.py
# Calculate sleep goal per day
def getSleepHours(day):
  day = day.lower()
  if day == "monday":
    return 8
  elif day == "tuesday":
    return 7
  elif day == "wednesday":
    return 6
  elif day == "thursday":
    return 5
  elif day == "friday":
    return 4
  elif day == "saturday":
    return 3
  elif day == "sunday":
    return 2
  else:
    return "Invalid input";

# Calculate how many hours were actually slept during each night
def getActualSleepHours():
  return getSleepHours("Monday") + getSleepHours("Tuesday") + getSleepHours("Wednesday") + getSleepHours("Thursday") + getSleepHours("Friday") + getSleepHours("Saturday") + getSleepHours("Sunday")
  
# Calculate the ideal number of hours slept per week
def getIdealSleepHours(idealHours):
  return idealHours * 7

# Compare ideal sleep and actual sleep
def calculateSleepDebt (ideal) :
  actualSleepHours = getActualSleepHours()
  idealSleepHours = getIdealSleepHours(ideal)
  if actualSleepHours == idealSleepHours:
    print("You get the perfect amount of sleep")
  elif actualSleepHours > idealSleepHours:
    totalDifference = actualSleepHours - idealSleepHours
    print("You get too much sleep")
    print("You are", str(totalDifference), "hours ahead of your schedule")
  else:
    totalDifference = idealSleepHours - actualSleepHours
    print("You don't get enough sleep")
    print("You are", str(totalDifference), "hours behind your schedule")
    
calculateSleepDebt(5);

